test/integration/controller/core/workload_controller_test.go

```diff
@@ -105,4 +107,40 @@ var _ = ginkgo.Describe("Workload controller", func() {
 			gomega.Expect(updatedQueueWorkload.Status.Conditions[0].Message).To(testing.Equal(message))
 		})
 	})
+
+	ginkgo.When("admit workload", func() {
+		ginkgo.BeforeEach(func() {
+			clusterQueue = testing.MakeClusterQueue("cluster-queue").
+				Resource(testing.MakeResource(resourceGPU).
+					Flavor(testing.MakeFlavor(flavorOnDemand, "5").Max("10").Obj()).Obj()).
+				Obj()
+			gomega.Expect(k8sClient.Create(ctx, clusterQueue)).To(gomega.Succeed())
+
+			queue = testing.MakeQueue("queue", ns.Name).ClusterQueue(clusterQueue.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, queue)).To(gomega.Succeed())
+		})
+		ginkgo.AfterEach(func() {
+			gomega.Expect(framework.DeleteQueue(ctx, k8sClient, queue)).To(gomega.Succeed())
+			gomega.Expect(framework.DeleteClusterQueue(ctx, k8sClient, clusterQueue)).To(gomega.Succeed())
+			gomega.Expect(framework.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+			updatedQueueWorkload = kueue.Workload{}
+			clusterQueue = nil
+		})
+
+		ginkgo.It("Should update status when workloads are admitted", func() {
+			ginkgo.By("Create workloads")
+			wl = testing.MakeWorkload("one", ns.Name).Queue(queue.Name).Request(corev1.ResourceCPU, "1").Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl)).To(gomega.Succeed())
+
+			ginkgo.By("Admit workloads")
+			gomega.Eventually(func() bool {
+				var newWL kueue.Workload
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(wl), &newWL)).To(gomega.Succeed())
+				newWL.Spec.Admission = testing.MakeAdmission(clusterQueue.Name).
+					Flavor(corev1.ResourceCPU, flavorOnDemand).Obj()
+				gomega.Expect(k8sClient.Update(ctx, &newWL)).To(gomega.Succeed())
```

Comment:
do the update once outside eventually; inside eventually we should only get the workload and check the condition